///////////////////////////////////////////////////////////////////////////
// Copyright (C) Flowbox, Inc / All Rights Reserved
// Unauthorized copying of this file, via any medium is strictly prohibited
// Proprietary and confidential
// Flowbox Team <contact@flowbox.io>, 2014
///////////////////////////////////////////////////////////////////////////

package generated.proto.parser;

import "dep/expr.proto";
import "dep/pat.proto";
import "dep/type.proto";



message Parse {
    message Expr {
        message Request   {
            required string str = 1;
        }
        message Status {
            required Request request = 1;
            required generated.proto.dep.expr.Expr expr = 2;
        }   
    }
    message Pat {
        message Request   {
            required string str = 1;
        }
        message Status {
            required Request request = 1;
            required generated.proto.dep.pat.Pat pat = 2;
        }   
    }
    message Type {
        message Request   {
            required string str = 1;
        }
        message Status {
            required Request request = 1;
            required generated.proto.dep.type.Type type = 2;
        }   
    }
    message NodeExpr {
        message Request   {
            required string str = 1;
        }
        message Status {
            required Request request = 1;
            required generated.proto.dep.expr.Expr expr = 2;
        }   
    }
}

message MkText {
    message Request {
        required generated.proto.dep.expr.Expr expr = 1;
    }
    message Status {
        required Request request = 1;
        required string  code    = 2;
    }
}

message Parser {
    message Ping {
        message Request { }
        message Status  { 
            required Request request = 1;
        }
    }
}
